hack/generated/controllers/generic_controller.go

```diff
@@ -225,10 +220,10 @@ func (gr *GenericReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, err
 	}
 
-	// TODO: This actually means we don't get exponential backoff which Kubernetes does by default,
-	// TODO: but we need this for test today to get fast reconciles.
-	if result.Requeue && result.RequeueAfter == time.Duration(0) {
+	// If we have a hardcoded requeue delay, force that
+	if gr.RequeueDelay != time.Duration(0) && (result.Requeue || result.RequeueAfter > time.Duration(0)) {
 		result.RequeueAfter = gr.RequeueDelay
+		result.Requeue = true
```

Comment:
`Requeue` is ignored if `RequeueAfter` is set… we should probably rethink this a bit, since we probably want the `controller-runtime` backoff…

Reply:
I think that this is keeping the `controller-runtime` backoff?

If the configured `gr.RequeueDelay` is `0` (the default in everything but the recording tests) then we don't do anything here and just return `result` as-is - which is what we want to do because that's what the underlying controller set.
If the configured `RequeueDelay` is nonzero (the tests) we just force the delay to be the specified delay, which yes deprives us of the backoff but that's what we want in the tests.

I think that's _better_ than what we had before because before the actual live behavior was that we were always setting `result.RequeueAfter = gr.RequeueAfter` whenever `result.RequeueAfter == 0` which was all the time because we don't set it in the controller?

Reply:
Possibly I need to extract the if parameters out into clearly named variables to make this more obvious though?

Reply:
I was misreading it a bit. But yes, renaming to something like `RequeueDelayOverride` would help!

Reply:
Changed this to `gr.RequeueDelayOverride` and also refactored the if statement to make it clearer.